src/drivers/amd/common/acp_dmic_dma.c

```diff
@@ -106,10 +107,40 @@ static int acp_dmic_dma_status(struct dma_chan_data *channel,
 static int acp_dmic_dma_copy(struct dma_chan_data *channel, int bytes,
 			     uint32_t flags)
 {
+	uint32_t i;
+	uint32_t j;
+	int numsamples;
+	char *dmic_rngbuff_addr2 = acp_initsilence.dmic_rngbuff_addr1;
 	struct dma_cb_data next = {
 		.channel = channel,
 		.elem.size = bytes,
 	};
+	if (acp_initsilence.silence_incr < acp_initsilence.silence_cnt) {
+		if (acp_initsilence.silence_incr % 2)
```

Comment:
we rely on the compiler to be "smart" enough to replace this with an `if (acp_initsilence.silence_incr & 1)` because we don't want to do easily avoidable divisions in `.copy()`

Reply:
Thanks @lyakh, updated the changes